tiledb/api/src/query/read/managed.rs

```diff
@@ -1,39 +1,164 @@
 use super::output::ScratchSpace;
 use super::*;
 
+pub enum ManagedScratch<'data> {
```

Comment:
This, and lots of other things, can be deleted by removing `register_constructor_managed` in favor of `ScratchStrategy`